testing/dsp_testing/test_beamforming.py

```diff
@@ -1,123 +1,693 @@
+
+    # find pulse points in data that is decimated. 
+
+    #plot_output_samples_iq_data(record_data[output_samples_filetype], output_samples_filetype)
+    #plot_bf_iq_data(record_data[bfiq_filetype], bfiq_filetype)
+
+    beamforming_dict = {}
+    print('BEAM AZIMUTHS: {}'.format(beam_azms))
+    for sequence_num in range(0,nave):
+        print('SEQUENCE NUMBER {}'.format(sequence_num))
+        sequence_dict = beamforming_dict[sequence_num] = {}
+        for filetype, record_dict in record_data.items():
+            #print(filetype)
+            sequence_filetype_dict = sequence_dict[filetype] = {}
+            data_description_list = list(record_dict['data_descriptors'])
+            # STEP 1: DECIMATE IF NECESSARY
+            if not math.isclose(record_dict['rx_sample_rate'], 3333.333, abs_tol=0.001):
+                # we aren't at 3.3 kHz - need to decimate.
+                #print(record_dict['rx_sample_rate'])
+                dm_rate = int(record_dict['rx_sample_rate']/3333.333)
+                #print(dm_rate)
+                dm_start_sample = record_dict['dm_start_sample']
+                dm_end_sample = -1 - dm_start_sample # this is the filter size 
+                if data_description_list == ['num_antenna_arrays', 'num_sequences', 'num_beams', 'num_samps']:
+                    decimated_data = record_dict['data'][0][sequence_num][:][dm_start_sample:dm_end_sample:dm_rate] # grab only main array data, first sequence, all beams.
+                    intf_decimated_data = record_dict['data'][1][sequence_num][:][dm_start_sample:dm_end_sample:dm_rate]
+                elif data_description_list == ['num_antennas', 'num_sequences', 'num_samps']:
+                    decimated_data = record_dict['data'][:,sequence_num,dm_start_sample:dm_end_sample:dm_rate] # first sequence only, all antennas.
+                elif data_description_list == ['num_sequences', 'num_antennas', 'num_samps']:
+                    if filetype == tx_filetype:  # tx data has sequence number 0 for all 
+                        decimated_data = record_dict['data'][0,:,dm_start_sample:dm_end_sample:dm_rate]
+                    else:
+                        decimated_data = record_dict['data'][sequence_num,:,dm_start_sample:dm_end_sample:dm_rate]
+                else:
+                    raise Exception('Not sure how to decimate with the dimensions of this data: {}'.format(record_dict['data_descriptors']))
+                
+            else:
+                if data_description_list == ['num_antenna_arrays', 'num_sequences', 'num_beams', 'num_samps']:
+                    decimated_data = record_dict['data'][0,sequence_num,:,:] # only main array
+                    intf_decimated_data = record_dict['data'][1,sequence_num,:,:]
+                elif data_description_list == ['num_antennas', 'num_sequences', 'num_samps']:
+                    decimated_data = record_dict['data'][:,sequence_num,:] # first sequence only, all antennas.
```

Comment:
does the comment "first sequence only" need to be changed to 'current sequence only'? or something